fusestore/store/src/executor/meta_handlers.rs

```diff
@@ -145,37 +207,83 @@ impl RequestHandler<DropTableAction> for ActionHandler {
         &self,
         act: DropTableAction,
     ) -> common_exception::Result<DropTableActionResult> {
-        let mut meta = self.meta.lock();
-        let _ = meta.drop_table(&act.plan.db, &act.plan.table, act.plan.if_exists)?;
-        Ok(DropTableActionResult {})
+        let db_name = &act.plan.db;
+        let table_name = &act.plan.table;
+        let if_exists = act.plan.if_exists;
+
+        let db = self.meta_node.get_database(db_name).await.ok_or_else(|| {
+            ErrorCode::UnknownDatabase(format!("drop table: database not found {:}", db_name))
+        })?;
```

Comment:
No need to get the db here either 🤔 
All the update must be done inside state machine.